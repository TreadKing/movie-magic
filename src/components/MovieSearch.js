import React, { useState } from 'react';
import Movie from './Movie.js';
import Watchlist from './Watchlist.js';


function MovieSearch(props) {

    const authToken = props.authToken
    const username = props.username

    const [searchKey, setSearchKey] = useState('')
    const [searchResults, setSearchResults] = useState([])
    const [switchToWatchlist, setSwitchToWatchlist] = useState(false)

    function getMovies() {
        setSearchKey('')

        const options = {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({
                'auth_token': authToken,
                'search_key': searchKey
            })
        }

        fetch('/search', options)
            .then(response => response.json())
            .then(result => setSearchResults(result))
    }

    function updateSearchKey(e) {
        setSearchKey(e.target.value)
    }

    function displayMovies() {
        const display = []
        for (var i = 0; i < searchResults.length; i++) {
            var movie = searchResults[i]
            display.push(<Movie
                authToken={authToken}
                movieTitle={movie['movie_title']}
                movieId={movie['movie_id']}
                movieImage={movie['movie_image']}
                rating={movie['rating']}
                status={movie['status']}
                comment={movie['comment']}
                authToken={authToken}
                key={i}
            />)
        }
        return <div className="movie-search-result">{display}</div>
    }

    if (switchToWatchlist) {
        return <Watchlist authToken={authToken}></Watchlist>
    } else {
        return <>
            <button onClick={() => setSwitchToWatchlist(true)}>Watchlist</button>
            <span className="search-input-container">
                <input placeholder="Enter an actor's name"
                    className="search-input"
                    type="text"
                    value={searchKey}
                    onChange={updateSearchKey} />
            </span>
            <button className="search-button" onClick={getMovies}>Search</button>
            {displayMovies()}
        </>
    }


}

export default MovieSearch;