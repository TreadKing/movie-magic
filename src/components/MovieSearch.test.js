import React from 'react';
import renderer from 'react-test-renderer';
import MovieSearch from './MovieSearch';

it('Test if the search page renders', () => {
  const component = renderer.create(<MovieSearch authToken=";)" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
