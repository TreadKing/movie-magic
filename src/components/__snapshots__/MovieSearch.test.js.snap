// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the search page renders 1`] = `
Array [
  <span
    className="menu"
  >
    <span
      className="watchlist-button-container"
    >
      <button
        className="watchlist-button"
        onClick={[Function]}
        type="button"
      >
        Logout
      </button>
    </span>
    <span
      className="watchlist-button-container"
    >
      <button
        className="watchlist-button"
        onClick={[Function]}
        type="button"
      >
        Watchlist
      </button>
    </span>
    <span
      className="upcoming-movies-button-container"
    >
      <button
        className="upcoming-movies-button"
        onClick={[Function]}
        type="button"
      >
        Upcoming
      </button>
    </span>
  </span>,
  <span
    className="search-form-container"
  >
    <span
      className="search-input-container"
    >
      <input
        className="search-input"
        onChange={[Function]}
        placeholder="Enter an actor's name"
        type="text"
        value=""
      />
    </span>
    <button
      className="search-button"
      onClick={[Function]}
      type="button"
    >
      Search
    </button>
    <p
      className="select-genre-container"
    >
      <span
        className="select-genre-label"
      >
        Genre
      </span>
      <select
        className="select-genre"
        onChange={[Function]}
      >
        <option
          value={null}
        />
        <option
          value="Action"
        >
          Action
        </option>
        <option
          value="Adventure"
        >
          Adventure
        </option>
        <option
          value="Animation"
        >
          Animation
        </option>
        <option
          value="Comedy"
        >
          Comedy
        </option>
        <option
          value="Crime"
        >
          Crime
        </option>
        <option
          value="Documentary"
        >
          Documentary
        </option>
        <option
          value="Drama"
        >
          Drama
        </option>
        <option
          value="Family"
        >
          Family
        </option>
        <option
          value="Fantasy"
        >
          Fantasy
        </option>
        <option
          value="History"
        >
          History
        </option>
        <option
          value="Horror"
        >
          Horror
        </option>
        <option
          value="Music"
        >
          Music
        </option>
        <option
          value="Mystery"
        >
          Mystery
        </option>
        <option
          value="Romance"
        >
          Romance
        </option>
        <option
          value="Science Fiction"
        >
          Science Fiction
        </option>
        <option
          value="TV Movie"
        >
          TV Movie
        </option>
        <option
          value="Thriller"
        >
          Thriller
        </option>
        <option
          value="War"
        >
          War
        </option>
        <option
          value="Western"
        >
          Western
        </option>
      </select>
    </p>
    <p
      className="input-year-container-2"
    >
      <span
        className="input-year-label-2"
      >
        Year
      </span>
      <select
        className="year-before-after-input-2"
        onChange={[Function]}
      >
        <option
          value={true}
        >
          before
        </option>
        <option
          value={false}
        >
          after
        </option>
      </select>
      <input
        className="year-input-2"
        onChange={[Function]}
        type="number"
      />
    </p>
    <p
      className="rating-container"
    >
      <span
        className="rating-label"
      >
        Rating
      </span>
      <select
        className="rating-less-greater-input"
        onChange={[Function]}
      >
        <option
          value={true}
        >
          less than
        </option>
        <option
          value={false}
        >
          greater than
        </option>
      </select>
      <select
        className="rating-input"
        onChange={[Function]}
      >
        <option
          value={null}
        />
        <option
          value={0}
        >
          0
        </option>
        <option
          value={1}
        >
          1
        </option>
        <option
          value={2}
        >
          2
        </option>
        <option
          value={3}
        >
          3
        </option>
        <option
          value={4}
        >
          4
        </option>
        <option
          value={5}
        >
          5
        </option>
        <option
          value={6}
        >
          6
        </option>
        <option
          value={7}
        >
          7
        </option>
        <option
          value={8}
        >
          8
        </option>
        <option
          value={9}
        >
          9
        </option>
        <option
          value={10}
        >
          10
        </option>
      </select>
    </p>
  </span>,
  <h1 />,
  <div
    className="movie-list"
  />,
]
`;
